### ISSUE TITLE
filter assumes "tags" is #tag and not meta data with key "tags"

### ISSUE THREAD

**Issue by tbm** [beancount/fava#685]
beancount allows you to specify meta data with key: value and the key "tags" is allowed.  This creates meta data with the key "tags".

You can also specify tags and they are stored differently.

Fava shows the tags and meta data differently.  #tags are displayed like this.  However, when searching for "tags" Fava assumes it's a #tag and doesn't take into consideration that there might be meta data with such a key.

I think the problem is in `fava/core/filters.py`:
```
        def _key(entry):
            if hasattr(entry, key):
                return match(str(getattr(entry, key) or ''))
            if key in entry.meta:
                return match(str(entry.meta.get(key)))
```

Test case:

```
option "operating_currency" "EUR"

2018-01-01 open Assets:Transfer
2018-01-01 open Assets:Current:Test

2018-03-12 * "Transfer 1"
     tags: "foo"
     Assets:Current:Test                    20.00 EUR
     Assets:Transfer                       -20.00 EUR
```

Go to http://localhost:5000/beancount/journal/ and the meta data is shown. Click on "foo" and the filter tags:"foo" is added: http://localhost:5000/beancount/journal/?filter=+tags%3A%22foo%22
However, the page is empty now.

Note: I filed a [bug on beancount](https://bitbucket.org/blais/beancount/issues/240/should-meta-data-with-key-tags-be-rejected) to ask if a key "tags" is allowed but since thinking about it some more after filing the bug I really don't see why it shouldn't be allowed.

**Comment by aumayr** [comment]
Thanks @tbm for reporting! 

Let's see what @blais [thinks about this](https://bitbucket.org/blais/beancount/issues/240/should-meta-data-with-key-tags-be-rejected) and discuss adaptions afterwards.

**Comment by yagebu** [comment]
What is the question/request here? The current behaviour seems to be as documented ('tags' is an entry attribute)

**Comment by tbm** [comment]
Correct, as @aumayr says.

And it's not just an assumption on my end.  The problem is that fava shows the meta data, allows me to click on it, but then generates a filter that doesn't actually look for this meta tag but for something else. (Hence the empty page)

**Comment by tbm** [comment]
Martin says it's valid.

**Comment by tbm** [comment]
So imho `tags:` should:

1. check if there's a tag on the account
2. check if there's a tag on posting (in the future, once this is allowed)
3. check if there's metadata with key 'tag'

**Comment by yagebu** [comment]
I disagree - it should only do 1. - 2. should be done explicitly and 3. shouldn't be encouraged IMHO.

I'll update the help page to make the current behaviour clearer.

**Comment by tbm** [comment]
I know you want filters to be universal but I think filters should depend on where they are called (i.e. if the filter is applied when looking at a specific account, it should look at the posting first and the transaction second).

But I'm very new to fava and don't understand the big picture, so maybe what I'm saying doesn't make sense.

Would be good to hear comments from @aumayr or other users too.

**Comment by tbm** [comment]
(btw, I agree that a metadata tag called "tags" shouldn't be encouraged, but it's a workaround because posting-level tags are currently not supported.)

**Comment by aumayr** [comment]
I agree with @yagebu - we should not encourage users to use "tags" as a metadata-key. Also, I think the filters should _not_ change behavior based on where you are on the page. Keeping them as general and "pure" as possible is the least ambiguous way, and that's how I think they should work.
